ToDoAppWithCoreData/Controllers/ListViewController.swift

```diff
@@ -0,0 +1,178 @@
+            tapBeginTimeButton(beginTimeButton)
+            
+        } else if !self.endDatePicker.isHidden {
+            
+            tapEndDateButton(endDateButton)
+            
+        } else if !self.endTimePicker.isHidden {
+            
+            tapEndTimeButton(endTimeButton)
+            
+        }
+        
+        UIView.animate(withDuration: 0.1, animations: {
+            self.beginDatePicker.isHidden = !self.beginDatePicker.isHidden
+        })
+        dateFormatter.setLocalizedDateFormatFromTemplate("yMMdE")
+        beginDateButton.setTitle("\(dateFormatter.string(from: beginDatePicker.date))", for: UIControl.State.normal)
+        
+    }
+    
+    @IBAction func tapBeginTimeButton(_ sender: UIButton) {
+        
+        if !self.beginDatePicker.isHidden {
+            tapBeginDateButton(beginDateButton)
+        } else if !self.endDatePicker.isHidden {
+            tapEndDateButton(endDateButton)
+        } else if !self.endTimePicker.isHidden {
+            tapEndTimeButton(endTimeButton)
+        }
+        UIView.animate(withDuration: 0.1, animations: {
+            self.beginTimePicker.isHidden = !self.beginTimePicker.isHidden
+        })
+        timeFormatter.setLocalizedDateFormatFromTemplate("jm")
+        beginTimeButton.setTitle("\(timeFormatter.string(from: beginTimePicker.date))", for: UIControl.State.normal)
+        
+    }
+    
+    @IBAction func tapEndDateButton(_ sender: UIButton) {
+        
+        if !self.beginDatePicker.isHidden {
+            tapBeginDateButton(beginDateButton)
+        } else if !self.beginTimePicker.isHidden{
+            tapBeginTimeButton(beginTimeButton)
+        } else if !self.endTimePicker.isHidden {
+            tapEndTimeButton(endTimeButton)
+        }
+        UIView.animate(withDuration: 0.1, animations: {
+            self.endDatePicker.isHidden = !self.endDatePicker.isHidden
+        })
+        dateFormatter.setLocalizedDateFormatFromTemplate("yMMdE")
+        endDateButton.setTitle("\(dateFormatter.string(from: endDatePicker.date))", for: UIControl.State.normal)
+        
+    }
+    
+    @IBAction func tapEndTimeButton(_ sender: UIButton) {
+        
+        if !self.beginDatePicker.isHidden {
+            tapBeginDateButton(beginDateButton)
+        } else if !self.beginTimePicker.isHidden{
+            tapBeginTimeButton(beginTimeButton)
+        } else if !self.endDatePicker.isHidden {
+            tapEndDateButton(endDateButton)
+        }
+        UIView.animate(withDuration: 0.1, animations: {
+            self.endTimePicker.isHidden = !self.endTimePicker.isHidden
+        })
+        dateFormatter.setLocalizedDateFormatFromTemplate("yMMdE")
+        endTimeButton.setTitle("\(timeFormatter.string(from: endTimePicker.date))", for: UIControl.State.normal)
+        
+    }
+    
+    override func touchesBegan(_ touches: Set<UITouch>, with event: UIEvent?) {
+        
+        self.view.endEditing(true)
+        if !self.beginDatePicker.isHidden {
```

Comment:
```suggestion
        if !beginDatePicker.isHidden {
```

この辺は好き嫌いですが、self無くても呼べます。